jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -522,16 +519,22 @@ ecma_create_named_data_property (ecma_object_t *obj_p, /**< object */
   ecma_property_t *prop_p = ecma_alloc_property ();
   name_p = ecma_copy_or_ref_ecma_string (name_p);
 
-  prop_p->type = ECMA_PROPERTY_NAMEDDATA;
+  prop_p->flags = ECMA_PROPERTY_FLAG_NAMEDDATA;
 
-  ECMA_SET_NON_NULL_POINTER (prop_p->u.named_data_property.name_p, name_p);
+  ECMA_SET_NON_NULL_POINTER (prop_p->v.named_data_property.name_p, name_p);
 
-  prop_p->u.named_data_property.writable = is_writable ? ECMA_PROPERTY_WRITABLE : ECMA_PROPERTY_NOT_WRITABLE;
-  prop_p->u.named_data_property.enumerable = is_enumerable ? ECMA_PROPERTY_ENUMERABLE : ECMA_PROPERTY_NOT_ENUMERABLE;
-  prop_p->u.named_data_property.configurable = (is_configurable ?
-                                                ECMA_PROPERTY_CONFIGURABLE : ECMA_PROPERTY_NOT_CONFIGURABLE);
-
-  prop_p->u.named_data_property.is_lcached = false;
+  if (is_configurable)
+  {
+    prop_p->flags = (uint8_t) (prop_p->flags | ECMA_PROPERTY_FLAG_CONFIGURABLE);
+  }
+  if (is_enumerable)
+  {
+    prop_p->flags = (uint8_t) (prop_p->flags | ECMA_PROPERTY_FLAG_ENUMERABLE);
+  }
+  if (is_writable)
+  {
+    prop_p->flags = (uint8_t) (prop_p->flags | ECMA_PROPERTY_FLAG_WRITABLE);
```

Comment:
Wouldn't `prop_p->flags |= (uint8_t) ECMA_PROPERTY_FLAG_xxx;` work here and above? (It's a bit more compact.)

Actually, using a local variable to compose the flags and then copy that into `prop_p->flags` would be even nicer.